### PR TITLE
Reduce number of calls to JsPool._metadata indexer to reduce synchron…

### PR DESCRIPTION
**Pull request by huan086**
…ization.

Fix concurrency issue with JsPool.EngineCount.
Closes Daniel15/JSPool#13

### CODE REVIEW

File: src/JSPool.Tests/JsPoolTests.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public class JsPoolTests
 		public void ConstructorCreatesEngines()
 		{
 			var factory = new Mock<IEngineFactoryForMock>();
-			factory.Setup(x => x.EngineFactory()).Returns(new Mock<IJsEngine>().Object);
+			factory.Setup(x => x.EngineFactory()).Returns(() => new Mock<IJsEngine>().Object);
```
Comment (Daniel15) [inline src/JSPool.Tests/JsPoolTests.cs]: Were these tests broken before? I haven't seen any failures, but maybe the tests weren't running :/

**Comment by Daniel15** [comment]
Thanks for the pull request! I'll test this out soon and merge it if everything looks good :)

File: src/JSPool.Tests/JsPoolTests.cs (hunk 1 above)
Comment (huan086) [inline src/JSPool.Tests/JsPoolTests.cs]: The metadata is keyed on the engine. The factory returns the same engine all the time. It works previously due to maintaining a separate engineCount, which did not take into account the uniqueness of the engine. It failed with change to use metadata.Count

Comment (Daniel15) [inline src/JSPool.Tests/JsPoolTests.cs]: Ohhh, good catch!

**Pull request merged**